src/main/java/io/stargate/sgv2/jsonapi/api/model/command/impl/CreateCollectionCommand.java

```diff
@@ -15,5 +14,39 @@ public record CreateCollectionCommand(
         @Size(min = 1, max = 48)
         @Pattern(regexp = "[a-zA-Z][a-zA-Z0-9_]*")
         @Schema(description = "Name of the collection")
-        String name)
-    implements NamespaceCommand, NoOptionsCommand {}
+        String name,
+    @Nullable
+        @Schema(
+            description = "Configuration for the collection",
+            type = SchemaType.OBJECT,
+            implementation = Options.class)
+        Options options)
+    implements NamespaceCommand {
+  public record Options(
+
+      // limit of returned documents
+      @Schema(
+              description = "Vector search index configuration for the collection",
+              type = SchemaType.OBJECT,
+              implementation = VectorSearchConfig.class)
+          VectorSearchConfig vector) {
+
+    public record VectorSearchConfig(
+        @Schema(
+                description = "Vector field embedding size",
+                type = SchemaType.INTEGER,
+                implementation = Integer.class)
+            Integer size,
```

Comment:
Should we add annotations to ensure this as required, and with minimum (1) and maximum (? do we know this) values?